### ISSUE TITLE
Automatically add labels if they're contained in .json files

### ISSUE THREAD

**Issue by andrewpap22** [HSF/phoenix#247]
Hello there,  

> Based on our discussion today on the meeting, I opened this issue 
(hopefully it doesn't make any problem because it was in your TODO, otherwise just delete it) 
and I'll try to work on it on these 4 days coming that I'll have more free time! 😄 

All the best, 

Andreas Pappas

**Comment by EdwardMoyse** [comment]
Hi Andreas, thanks for doing this! An example of a file with labels is:
https://github.com/HSF/phoenix/pull/251/commits/8e6ffcb23c424d0e702a2c9eb0595be9fe33a20d

**Comment by Pratz2005** [comment]
Hello, can I work on this issue

**Comment by EdwardMoyse** [comment]
Yes, you can (in general you are free to work on any issue which hasn't been assigned to someone)

**Comment by Pratz2005** [comment]
Ok will do thank you

**Comment by Pratz2005** [comment]
Hello, are the json files being changed during runtime?

**Comment by EdwardMoyse** [comment]
No, these will not change during runtime.

**Comment by Pratz2005** [comment]
I have added the labels as eta/phi for all objects in the atlas.json file but for the tracks, I'm a bit confused on how to go about the labelling can you help me with that

**Comment by Pratz2005** [comment]
Hi @EdwardMoyse can I get an update on this?

**Comment by EdwardMoyse** [comment]
Hi @Pratz2005 - sorry, I have been offline. Can you point me to your code? 

Basically I would think you would follow the logic in 
[  private loadLabelsFile() {
](https://github.com/HSF/phoenix/blob/06827f4858e5930601775395774e52f3f3536dee/packages/phoenix-event-display/src/managers/ui-manager/index.ts#L564)

**Comment by Pratz2005** [comment]
![image](https://github.com/user-attachments/assets/3c5d8ae4-c7f0-452d-b62e-aed3adf68318)
So how I understood the problem was that in the json files there are no labels so I just wrote a python script to parse through the atlas.json file to add labels as eta/phi for all objects which have eta and phi as variables as shown in the example.

[atlas.json](https://github.com/HSF/phoenix/blob/main/packages/phoenix-ng/projects/phoenix-app/src/assets/files/event_data/atlas.json)

**Comment by EdwardMoyse** [comment]
Aaah! No, sorry - that's not it. I don't care about what the actual content of the labels is - they are just placeholders. 

What we want to do is to streamline adding labels to objects i.e. instead of this taking two steps:
- loading an event data file (e.g. `atlas.json`, but it should work for all formats), 
- and then loading a `labels.json` file.

It should be one step: open the event data file, and then if the event data objects have a _label_ attribute the we would automatically call `this.three.addLabelToObject` to put the labels into the 3D view.

**Comment by Pratz2005** [comment]
![image](https://github.com/user-attachments/assets/e6534911-aae5-4d25-ad1f-a70521bb437e)

Hi @EdwardMoyse can you assist me on how to test the issue on the UI. Do you want me to add labels into the view 
 when i import the [ATLAS_calibration.json](https://github.com/HSF/phoenix/blob/main/packages/phoenix-ng/projects/phoenix-app/src/assets/files/event_data/ATLAS_calibration.json) file?

**Comment by Pratz2005** [comment]
@EdwardMoyse Can I get an update on this
